src/mesh/Data.cpp

```diff
@@ -13,7 +14,9 @@ size_t Data::_dataCount = 0;
 Data::Data()
     : _name(""),
       _id(-1),
-      _dimensions(0)
+      _dimensions(0),
+      _meshDimensions(-1),
+      _hasGradient(false)
```

Comment:
You have quite some repetition in the constructors here and I think you should use a constructor delegation. In principle, you can also assign default values within the class declaration.